include/utility/common_utility.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+#include "logger.hpp"
+
+#include <vector>
+
+/**
+ * @brief Namespace to host common utility methods.
+ *
+ * A method qualifies as a common utility function if.
+ * A)It is being used by the utility namespace at the same level as well as
+ * other files directly.
+ * B) The utility should be a leaf node and should not be dependent on any other
+ * utility.
+ *                  *******************
+ *                  | Commmon Utility | - - - - - - -
+ *                  *******************              |
+ *                          /\                       |
+ *                         /  \                      |
+ *         ****************    ****************      |
+ *         | json utility |    | dbus utility |      |
+ *         ****************    ****************      |
+ *                 \                 /               |
+ *                  \               /                |
+ *                  *******************              |
+ *                  | Generic Utility | - - - - - - -
```

Comment:
sorry for my ignorance, but what is the difference between generic and common utility?

Reply:
we would need to update the generic name.